SU-HSDO/suhumsci
# STN-522 : Utility Class to Display Helper Link Text

# READY FOR REVIEW

## Summary
Adds a new utility class of `hb-display-more-link-text` that will unhide the helper link text in cards when added to views, and postcards when added to layouts in the Colorful theme.

### Acceptance Criteria
- [x] The utility class should have no effect on the Traditional theme since the link text is already visible.
- [x] There should be no changes to the default state of the .hb-card__decorative-link element. Changes should only happen when the utility class has been added.
- [x] Confirm the text styles and any spacing between the text and arrow with a designer. Should look similar to the more link styling on Colorful.
- [x] Links and link text should be accessible both with and without the utility class.
- [x] Browser test in all modern browsers and IE 11.

## Need Review By (Date)
8/24/2020

## Urgency
medium

## Steps to Test
1. In your CLI, run `npm run test` and confirm our tests pass.
2. To review the work in this PR, set your theme to `Colorful`.
3. Test unhiding the helper text in postcards.
    * Go to http://sparkbox-sandbox.suhumsci.loc/qa/qa-postcards
    * In the Drupal admin toolbar, click on `Layout`
    * Locate the postcards block in the center column of `Section 2` below the `QA Postcards` block. Click on the contextual link and select `Configure`
    * Add the utility class of `hb-display-more-link-text` to the `Field Formatter Class` then click `Update`
    * `Save layout`
   * Review QA Postcards page and confirm the arrow link in postcards has updated to display the helper text and has the more link styling for Colorful. See the WYSIWYG more link styling examples on http://sparkbox-sandbox.suhumsci.loc/qa/qa-page as a reference.
4. Test unhiding the helper text in cards in views.
    * Go to http://sparkbox-sandbox.suhumsci.loc/qa-views
    * Find the contextual link for the date stacked vertical event cards and click `Edit view`
    * Under the `Advanced` column, find `CSS class`, add `hb-display-more-link-text`, and `Save` the view
    * Review QA Views page and confirm the arrow link in the event cards has updated to display the helper text and has the more link styling for Colorful.
5. Switch to the Traditional theme and review the QA Postcards and QA Views pages. Confirm the styles on Traditional have not changed since the helper text is already visible on the Traditional theme.
